Add license qualifier section to code search syntax (#43285)
github/docs: content/search-github/github-code-search/understanding-github-code-search-syntax.md

For a complete list of supported language names, see [languages.yaml](https://github.com/github-linguist/linguist/blob/main/lib/linguist/languages.yml) in [github-linguist/linguist](https://github.com/github-linguist/linguist). If your preferred language is not on the list, you can open a pull request to add it.

### License qualifier

> You can filter repositories based on their license or license family using the `license:` qualifier and the exact license keyword, e.g. `Apache-2.0`, `CC`, `MIT`.

Reviewer: ```suggestion
To filter repositories based on their license or license family, use the `license:` qualifier and the exact license keyword, for example `Apache-2.0`, `CC`, `MIT`.
```
